pkg/component/worker/static_pods.go

```diff
@@ -0,0 +1,535 @@
+	if pod.APIVersion != "v1" || pod.Kind != "Pod" {
+		return fmt.Errorf("not a Pod: %s/%s", pod.APIVersion, pod.Kind)
+	}
+
+	// Validate that the pod is matching this claim.
+	if id := (staticPodID{pod.Namespace, pod.Name}); id != p.staticPodID {
+		return fmt.Errorf("attempt to set the manifest to %q, whereas %q was claimed", &id, &p.staticPodID)
+	}
+
+	// Update this pod's content, if not already dropped.
+	p.mu.Lock()
+	update := p.update
+	if update == nil {
+		p.mu.Unlock()
+		return errors.New("already dropped")
+	}
+	p.manifestPtr.Store(jsonBytes)
+	p.mu.Unlock()
+
+	// Update the content of the enclosing staticPods instance, without holding
+	// this pod's lock, so that there's no potential deadlocks. The update
+	// method itself will check if the staticPods instance has been stopped
+	// concurrently, anyways.
+	update()
+	return nil
+}
+
+func (p *staticPod) Clear() {
+	// Clear this pod's content.
+	p.mu.Lock()
+	update := p.update
+	p.manifestPtr.Store([]byte{})
+	p.mu.Unlock()
+
+	// If this pod hasn't been dropped already, update the content of the
+	// enclosing staticPods instance. Do this without holding this pod's lock,
+	// so that there's no potential deadlocks. The update method itself will
+	// check if the staticPods instance has been stopped concurrently, anyways.
+	if update != nil {
+		update()
+	}
+}
+
+func (p *staticPod) Drop() {
+	// Clear this pod's content, and unhook it from its enclosing staticPods instance.
+	p.mu.Lock()
+	drop := p.drop
+	p.update = nil
+	p.drop = nil
+	p.manifestPtr.Store([]byte{})
+	p.mu.Unlock()
+
+	// If this pod hasn't been dropped already, drop it from the enclosing
+	// staticPods instance. Do this without holding this pod's lock, so that
+	// there's no potential deadlocks. The drop method will check if the
+	// staticPods instance has been stopped concurrently, anyways.
+	if drop != nil {
+		drop()
+	}
+}
+
+// String returns a loggable representation for staticPodIds.
+func (i *staticPodID) String() string {
+	return fmt.Sprintf("%s/%s", i.namespace, i.name)
+}
+
+// update regenerates the content and stores it.
+func (s *staticPods) update() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Don't update anything if this instance has been stopped already.
+	if s.peekLifecycle() >= staticPodsStopped {
+		return
+	}
+
+	s.contentPtr.Store(generateContent(s.claimedPods))
+}
+
+// drop removes the given id, regenerates the content and stores it.
+func (s *staticPods) drop(id staticPodID) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// No need to drop anything if this instance has been stopped already.
+	if s.peekLifecycle() >= staticPodsStopped {
+		return
+	}
+
+	delete(s.claimedPods, id)
+	s.contentPtr.Store(generateContent(s.claimedPods))
+}
+
+// generateContent returns a JSON encoded list of pods, to be consumed by the kubelet.
+func generateContent(pods map[staticPodID]*staticPod) []byte {
+	var buf bytes.Buffer
+
+	buf.WriteString(`{"apiVersion":"v1","kind":"List","items":[`)
```

Comment:
As discussed, this needs to be a `PodList` rather than a generic `List`

Reply:
Done.